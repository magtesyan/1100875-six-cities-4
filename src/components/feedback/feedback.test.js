import renderer from "react-test-renderer";
import React from "react";
import shortid from "shortid";

import Feedback from "./feedback.jsx";

const feedback = {
  id: 1,
  comment: `Everything. It was really really nice to get away and feel so comfortable where we was. My kids and I had a great time.`,
  rating: 5,
  date: `2019-05-08T14:13:56.569Z`,
  user: {
    avatar_url: `img/1.png`,
    id: 4,
    is_pro: true,
    name: `Alex`
  }
};

it(`Feedbacks are rendered correctly`, () => {
  const tree = renderer
    .create(
        <Feedback
          key={shortid.generate()}
          feedback={feedback}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
