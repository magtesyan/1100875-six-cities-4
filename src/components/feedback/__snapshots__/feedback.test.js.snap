// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedbacks are rendered correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="img/1.png"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Alex
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      Everything. It was really really nice to get away and feel so comfortable where we was. My kids and I had a great time.
    </p>
    <time
      className="reviews__time"
      dateTime="May 8, 2019"
    >
      May 8, 2019
    </time>
  </div>
</li>
`;
